Description: Add additional console for aaarch64.
Author: Chuck Short <zulcss@ubuntu.com>
Fowarded: Not-needed
diff --git a/nova/virt/libvirt/driver.py b/nova/virt/libvirt/driver.py
index 66366c4..e988b89 100644
--- a/nova/virt/libvirt/driver.py
+++ b/nova/virt/libvirt/driver.py
@@ -3020,7 +3020,7 @@ class LibvirtDriver(driver.ComputeDriver):
         inst_path = libvirt_utils.get_instance_path(instance)
         disk_mapping = disk_info['mapping']
 
-        CONSOLE = "console=tty0 console=ttyS0"
+        CONSOLE = "console=tty0 console=ttyS0 console=ttyAM0"
 
         guest = vconfig.LibvirtConfigGuest()
         guest.virt_type = CONF.libvirt.virt_type
