diff -Naurp nova-2014.1.b2.orig/nova/cmd/novncproxy.py nova-2014.1.b2/nova/cmd/novncproxy.py
--- nova-2014.1.b2.orig/nova/cmd/novncproxy.py	2014-01-23 10:06:46.000000000 -0500
+++ nova-2014.1.b2/nova/cmd/novncproxy.py	2014-01-31 08:43:16.667524506 -0500
@@ -77,8 +77,6 @@ def main():
                                    daemon=CONF.daemon,
                                    record=CONF.record,
                                    web=CONF.web,
-                                   file_only=True,
-                                   no_parent=True,
                                    target_host='ignore',
                                    target_port='ignore',
                                    wrap_mode='exit',
