diff -Naurp nova-2013.1.a4257.g4df4762.orig/tools/hacking.py nova-2013.1.a4257.g4df4762/tools/hacking.py
--- nova-2013.1.a4257.g4df4762.orig/tools/hacking.py	2013-02-12 07:33:18.000000000 -0600
+++ nova-2013.1.a4257.g4df4762/tools/hacking.py	2013-02-12 07:51:45.212644383 -0600
@@ -589,7 +589,7 @@ if __name__ == "__main__":
     #include nova path
     sys.path.append(os.getcwd())
     #Run once tests (not per line)
-    once_error = once_git_check_commit_title()
+    #once_error = once_git_check_commit_title()
     #NOVA error codes start with an N
     pep8.SELFTEST_REGEX = re.compile(r'(Okay|[EWN]\d{3}):\s(.*)')
     pep8.ERRORCODE_REGEX = re.compile(r'[EWN]\d{3}')
