Description: Temporarily ignore pep8 w602 errors.
Author: Chuck Short <zulcss@ubuntu.com>
Forwarded: No
diff -Naurp nova-2012.2.orig/run_tests.sh nova-2012.2/run_tests.sh
--- nova-2012.2.orig/run_tests.sh	2012-05-02 10:59:59.000000000 -0400
+++ nova-2012.2/run_tests.sh	2012-05-03 08:59:25.657554551 -0400
@@ -110,7 +110,7 @@ function run_pep8 {
   #
 
   # Until all these issues get fixed, ignore.
-  ignore='--ignore=N4,N306'
+  ignore='--ignore=N4,N306,W602'
   ${wrapper} python tools/hacking.py ${ignore} ${srcfiles}
 }
 
