diff -Naurp nova-2012.2.orig/bin/nova-manage nova-2012.2/bin/nova-manage
--- nova-2012.2.orig/bin/nova-manage	2012-05-30 20:15:03.000000000 -0400
+++ nova-2012.2/bin/nova-manage	2012-05-31 14:23:51.564889384 -0400
@@ -1669,7 +1669,7 @@ def methods_of(obj):
 
 def main():
     """Parse options and call the appropriate class/method."""
-    rpc.register_opts(FLAGS)
+    rpc.register_opts('/etc/nova/nova.conf')
 
     try:
         argv = flags.parse_args(sys.argv)
