diff -Naurp nova.orig/bin/nova-manage nova/bin/nova-manage
--- nova.orig/bin/nova-manage	2012-02-06 09:42:20.625153598 -0500
+++ nova/bin/nova-manage	2012-02-06 09:46:10.141153721 -0500
@@ -2307,7 +2307,7 @@ def methods_of(obj):
 
 def main():
     """Parse options and call the appropriate class/method."""
-    flagfile = utils.default_flagfile()
+    flagfile = utils.default_flagfile('/etc/nova/nova.conf')
 
     if flagfile and not os.access(flagfile, os.R_OK):
         st = os.stat(flagfile)
