Description: Remove deprecated sqlalchemy assert_unicode.
Author: Chuck Short <zulcss@ubuntu.com>
Forwarded: Yes (https://review.openstack.org/#/c/39296/)
diff -Naurp nova-2013.2.a2367.gb404bfd.orig/nova/db/sqlalchemy/migrate_repo/versions/203_make_user_quotas_key_and_value.py nova-2013.2.a2367.gb404bfd/nova/db/sqlalchemy/migrate_repo/versions/203_make_user_quotas_key_and_value.py
--- nova-2013.2.a2367.gb404bfd.orig/nova/db/sqlalchemy/migrate_repo/versions/203_make_user_quotas_key_and_value.py	2013-07-30 13:01:25.000000000 +0000
+++ nova-2013.2.a2367.gb404bfd/nova/db/sqlalchemy/migrate_repo/versions/203_make_user_quotas_key_and_value.py	2013-07-30 13:44:02.410948311 +0000
@@ -35,7 +35,7 @@ def upgrade(migrate_engine):
     quota_usages = utils.get_table(migrate_engine, 'quota_usages')
     user_id = Column('user_id',
                      String(length=255, convert_unicode=False,
-                            assert_unicode=None, unicode_error=None,
+                            unicode_error=None,
                             _warn_on_bytestring=False))
     quota_usages.create_column(user_id)
 
@@ -43,7 +43,7 @@ def upgrade(migrate_engine):
                                 db._SHADOW_TABLE_PREFIX + 'quota_usages')
     user_id = Column('user_id',
                      String(length=255, convert_unicode=False,
-                            assert_unicode=None, unicode_error=None,
+                            unicode_error=None,
                             _warn_on_bytestring=False))
     shadow_quota_usages.create_column(user_id)
 
@@ -51,7 +51,7 @@ def upgrade(migrate_engine):
     reservations = utils.get_table(migrate_engine, 'reservations')
     user_id = Column('user_id',
                      String(length=255, convert_unicode=False,
-                            assert_unicode=None, unicode_error=None,
+                            unicode_error=None,
                             _warn_on_bytestring=False))
     reservations.create_column(user_id)
 
@@ -59,7 +59,7 @@ def upgrade(migrate_engine):
                                 db._SHADOW_TABLE_PREFIX + 'reservations')
     user_id = Column('user_id',
                      String(length=255, convert_unicode=False,
-                            assert_unicode=None, unicode_error=None,
+                            unicode_error=None,
                             _warn_on_bytestring=False))
     shadow_reservations.create_column(user_id)
 
@@ -83,17 +83,17 @@ def upgrade(migrate_engine):
                         Column('deleted', Integer),
                         Column('user_id',
                                String(length=255, convert_unicode=False,
-                                      assert_unicode=None, unicode_error=None,
+                                      unicode_error=None,
                                       _warn_on_bytestring=False),
                                nullable=False),
                         Column('project_id',
                                String(length=255, convert_unicode=False,
-                                      assert_unicode=None, unicode_error=None,
+                                      unicode_error=None,
                                       _warn_on_bytestring=False),
                                nullable=False),
                         Column('resource',
                                String(length=255, convert_unicode=False,
-                                      assert_unicode=None, unicode_error=None,
+                                      unicode_error=None,
                                       _warn_on_bytestring=False),
                                nullable=False),
                         Column('hard_limit', Integer, nullable=True),
