Description: Add missing nova-conductor.
Author: Chuck Short <zulcss@ubuntu.com>
Forwarded: Yes
diff -Naurp nova-2013.1.orig/setup.py nova-2013.1/setup.py
--- nova-2013.1.orig/setup.py	2012-11-19 20:58:05.000000000 -0600
+++ nova-2013.1/setup.py	2012-11-20 08:32:27.196490097 -0600
@@ -53,6 +53,7 @@ setuptools.setup(name='nova',
                'bin/nova-cert',
                'bin/nova-clear-rabbit-queues',
                'bin/nova-compute',
+               'bin/nova-conductor',
                'bin/nova-console',
                'bin/nova-consoleauth',
                'bin/nova-dhcpbridge',
