Index: nova-2012.1~e4~20120207.12506/nova/rootwrap/filters.py
===================================================================
--- nova-2012.1~e4~20120207.12506.orig/nova/rootwrap/filters.py
+++ nova-2012.1~e4~20120207.12506/nova/rootwrap/filters.py
@@ -101,6 +101,7 @@
 
     def match(self, userargs):
         args = list(userargs)
+        print 'args', args
         if len(args) == 3:
             signal = args.pop(1)
             if signal not in self.args[0]:
@@ -115,11 +116,14 @@
                 return False
         pid = int(args[1])
         try:
+            print 'doing os.readlink("/proc/%d/exe"' % pid
             command = os.readlink("/proc/%d/exe" % pid)
             if command not in self.args[1]:
+                print 'denied'
                 # Affected executable not in accepted list
                 return False
         except:
+            print 'failed'
             # Incorrect PID
             return False
         return True
