Description: Use the oslo.sphinx namespace.
Author: Chuck Short <zulcss@ubuntu.com>
Fowarded: Not needed.
diff --git a/doc/source/conf.py b/doc/source/conf.py
index 0ae450f..95126b6 100644
--- a/doc/source/conf.py
+++ b/doc/source/conf.py
@@ -34,7 +34,7 @@ extensions = ['sphinx.ext.autodoc',
               'sphinx.ext.pngmath',
               'sphinx.ext.ifconfig',
               'sphinx.ext.graphviz',
-              'oslosphinx',
+              'oslo.sphinx',
               ]
 
 todo_include_todos = True
diff --git a/test-requirements.txt b/test-requirements.txt
index 76854a6..32b0af2 100644
--- a/test-requirements.txt
+++ b/test-requirements.txt
@@ -10,6 +10,6 @@ psycopg2
 pylint==0.25.2
 python-subunit
 sphinx>=1.1.2,<1.2
-oslosphinx
+oslo.sphinx
 testrepository>=0.0.17
 testtools>=0.9.32

